src/app/embed/embed.component.ts

```diff
@@ -1,15 +1,18 @@
 import { Component, OnInit } from '@angular/core';
 import { ActivatedRoute } from '@angular/router';
-import { EMBED_AUDIO_URL_PARAM, EMBED_TITLE_PARAM, EMBED_SUBTITLE_PARAM,
-  EMBED_SUBSCRIBE_URL_PARAM, EMBED_SUBSCRIBE_TARGET, EMBED_IMAGE_URL_PARAM } from './embed.constants';
+import { MergeAdapter } from './adapters/merge.adapter';
+import { QSDAdapter } from './adapters/qsd.adapter'
+import { FeedAdapter } from './adapters/feed.adapter'
+import { AdapterProperties } from './adapters/adapter.properties'
 
 @Component({
   selector: 'play-embed',
   styleUrls: ['embed.component.css'],
+  providers: [MergeAdapter, QSDAdapter, FeedAdapter],
```

Comment:
Hmm... I'm surprised angular lets you get away with this.  At some point in angular2 final, they moved `providers` to the Module level.  It usually yells at me if I try to include them in a Component.